src/main/scala/com/databricks/spark/redshift/RedshiftRelation.scala

```diff
@@ -136,13 +140,12 @@ private[redshift] case class RedshiftRelation(
     val query = {
       // Since the query passed to UNLOAD will be enclosed in single quotes, we need to escape
       // any single quotes that appear in the query itself
-      val tableNameOrSubquery: String = {
-        val unescaped = params.query.map(q => s"($q)").orElse(params.table.map(_.toString)).get
-        unescaped.replace("'", "\\'")
-      }
-      s"SELECT $columnList FROM $tableNameOrSubquery $whereClause"
+      val escapedTableNameOrSubqury = tableNameOrSubquery.replace("'", "\\'")
+      s"SELECT $columnList FROM $escapedTableNameOrSubqury $whereClause"
     }
-    val fixedUrl = Utils.fixS3Url(tempDir)
+    // We need to remove S3 credentials from the unload path URI because they will conflict with
+    // the credentials passed via `credsString`.
+    val fixedUrl = Utils.fixS3Url(Utils.removeCredentialsFromURI(new URI(tempDir)).toString)
```

Comment:
If you don't do this, Redshift will complain. In the log message, Redshift will actually show the credentials in its error message, which it won't do if they're in `credsString`, so this also fixes a minor security problem.